### PR TITLE
add DCOS version to `dcos --version`

### PR DESCRIPTION
**Pull request by tamarrow**
#567 
screen shot 2016-04-15 at 2 42 59 pm

cc @jsancio

### CODE REVIEW

File: cli/dcoscli/main.py [hunk 1]
```diff
@@ -20,12 +23,36 @@ def main():
...
+    :param config: config
+    :type config: Toml
+    :returns: versions of DCOS and DCOS-CLI
+    :rtype: dict
+    """
+
+    dcos_url = config.get("core.dcos_url")
+    dcos_version = "N/A"
+    if dcos_url:
+        url = urllib.parse.urljoin(dcos_url, 'dcos-metadata/dcos-version.json')
```
Comment (jsancio) [inline cli/dcoscli/main.py]: Why not use `dcos.http`?

File: cli/dcoscli/main.py [hunk 2]
```diff
@@ -20,12 +23,36 @@ def main():
         return 1
 
 
+def _get_versions(config):
```
Comment (jsancio) [inline cli/dcoscli/main.py]: Why not pass `dcos_url`? It is the only thing that we use from the config file.

**Comment by jsancio** [comment]
@tamarrow Can you include the sample output of `dcos --version` in the description for this PR?

File: cli/dcoscli/main.py [hunk 3]
```diff
@@ -20,12 +23,36 @@ def main():
...
+    :returns: versions of DCOS and DCOS-CLI
+    :rtype: dict
+    """
+
+    dcos_url = config.get("core.dcos_url")
+    dcos_version = "N/A"
+    if dcos_url:
+        url = urllib.parse.urljoin(dcos_url, 'dcos-metadata/dcos-version.json')
+        try:
+            res = http.get(url)
```
Comment (tamarrow) [inline cli/dcoscli/main.py]: @jsancio using `dcos.http` library. ^above is just to create the url

**Comment by jsancio** [comment]
@tamarrow thanks for the output. Why don't we turn this until more human friendly output like:

```
dcos --version
dcos-cli/0.4.3 dcos/1.7.0
```

I am worry that `dcos/1.7.0` may not be enough information when running against `testing/continuous` builds. Is there a way to get and display the git commit sha in addition to the version? cc @cmaloney

**Comment by tamarrow** [comment]
I wanted to make it scriptable, but since its mostly used for debugging, a string is fine. I can also add the dcos-image commit, which is accessable by that same endpoint.
So something like: `dcos-cli/0.4.3 dcos/1.7.0/dcos-image/hash-here`
cc @jsancio

**Comment by cmaloney** [comment]
I'd really rather not have another format for versions of the dcos build... Could we just output "dcos/version/<version>"
"dcos/commit/<dcos-image-commit>"
"dcos/bootstrap-id/<bootstrap-id>"
all as three different things?

**Comment by tamarrow** [comment]
@jsancio updated with `dcos config show` style formatting: 
screen shot 2016-04-18 at 4 51 23 pm

**Comment by jsancio** [comment]
LGTM. :shipit:

File: cli/dcoscli/main.py [hunk 4]
```diff
@@ -20,12 +22,44 @@ def main():
...
+        except Exception:
+            pass
+
+    return (
+        "dcoscli.version={}\n".format(dcoscli.version) +
+        "dcos.version={}\n".format(dcos_info.get("version", "N/A")) +
+        "dcos.commit={}\n".format(dcos_info.get(
+            "dcos-image-commit", "N/A")) +
+        "dcos.bootstrap-id={}".format(dcos_info.get("bootstrap-id", "N/A"))
+    )
```
Comment (jsancio) [inline cli/dcoscli/main.py]: Okay for now. Let's file an issue to use the same code we wrote for `dcos config show`.

**Pull request merged**